app/src/main/java/com/example/sharingang/items/FirestoreItemRepository.kt

```diff
@@ -101,7 +101,7 @@ class FirestoreItemRepository @Inject constructor() :
         requireNotNull(item.id)
 
         return try {
-            firestore.collection(collectionName).document(item.id!!)
+            firestore.collection(collectionName).document(item.id)
```

Comment:
When a value is nullable (for exemple String?), we cannot use it directly with something that accepts a non-nullable type. We can either check if it is not null then use it, or if we are really sure it is not null, we can use `!!` to cast to non-nullable (but it throws an exception if it contains a null value).
Here because we have requireNotNull(item.id), Kotlin understands that the value has been checked so the !! is not required.